### DISCUSSION TITLE
Unable to use callback URL and Cloudflare tunnel

### DISCUSSION THREAD

**Discussion by lehno** [jbergler/hass-ttlock#92] [Q&A]
When I try to add my webhook url, that is using Cloudflare tunnels, it shows the following message:

Callback url test failed:javax.net.ssl.SSLHandshakeException : sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

I know it is not related to the app here itself, but maybe someone knows how to fix it?

**Comment by jbergler** [comment]
I haven't personally used Cloudflare tunnels but I'd guess that this error might be cloudflare issuing certs from a newer certificate authority which ttlocks cloud doesn't know about.

You could try emailing ttlock-developers-email-list@googlegroups.com with the error (along with the hostname/url you're having issues with) and see if they're devs respond?

**Comment by nkiest** [comment]
Before I go down this rabbit hole myself, has anyone gotten this to work with Cloudflare tunnels?

**Comment by FragRot** [comment]
I received the same error when I use wireguard tunnel. Anyone managed to resolve this error? I'm using self-signed certificate.

**Comment by hettinger77** [comment]
same error here with ttlock adn cloudflare tunnel
